redbox/models/base.py

```diff
@@ -1,14 +1,18 @@
 from datetime import datetime
-from typing import Optional
+from typing import Annotated, Optional
 from uuid import UUID, uuid4
 
-from pydantic import BaseModel, Field, computed_field
+from pydantic import AfterValidator, BaseModel, Field, computed_field
 
 
 class PersistableModel(BaseModel):
-    uuid: UUID = Field(default_factory=uuid4)
+    uuid: UUID | Annotated[str, AfterValidator(lambda x: UUID(x, version=4))] = Field(
```

Comment:
v nice, i like this approach, one minor suggestion:

```suggestion
    uuid: UUID | Annotated[str, AfterValidator(UUID)] = Field(
```
as the version doesnt apply in this case where we are just casting it?

Reply:
I think the default factory is still useful here as the generator for when nothing is passed in. Both validation approaches look valid to me, shall leave it up to your taste!